### ISSUE TITLE
Test use of permission groups

### ISSUE THREAD

**Issue by derrell** [app-inventor-gallery/aig#38]
There is code in the MDbifCommon authorization function to handle authorizing an RPC based on a user's membership in a permission group.  That code is commented out. It has typos and has clearly never been tested. Correct this code, and then write unit tests to validate that permission groups are working correctly.

**Comment by cpuwhiz11** [comment]
From Derrell's description: 

specifically, you want to ensure (a) if a user does not have an explicit permission for an rpc and is not in a permission group for an rpc, that he can not execute the rpc; (b) if he has explicit permissions to an rpc but is not in a pgroup that would give him permission for an rpc, that he can execute the rpc; (c) if he is in a pgroup that gives permission to execute the rpc but does not have the explicit permission, that he can execute the rpc; (d) if he has both explicit permission and is in a pgroup that provides permission, that he can execute the rpc; and (e) if he is in multiple pgroups that provide the permission to an rpc that he can execute the rpc.
in other words, if he is supposed to have permission to an rpc he can execute it; otherwise he absolutely can NOT execute it.
(a) and (b) have existing tests that you can begin with. they're all quite old, though, and need to be confirmed to adequately test what we're talking about.

**Comment by derrell** [comment]
@cpuwhiz11 are all of the tests you intended to implement complete?

**Comment by cpuwhiz11** [comment]
Correct.

**Comment by derrell** [comment]
Closing.
